apiserver/facades/client/bundle/bundle.go

```diff
@@ -151,10 +156,111 @@ func (b *BundleAPI) GetChanges(args params.BundleChangesParams) (params.BundleCh
 }
 
 // ExportBundle exports the current model configuration as bundle.
-func (b *BundleAPI) ExportBundle() (params.StringResult, error) {
-	return params.StringResult{}, nil
+func (b *BundleAPI) ExportBundle() (params.BytesResult, error) {
+	if err := b.checkCanRead(); err != nil {
+		return params.BytesResult{}, common.ServerError(common.ErrPerm)
+	}
+
+	var exportConfig state.ExportConfig
+
+	// Skip attributes which are not required.
+	exportConfig.SkipActions = true
+	exportConfig.SkipCloudImageMetadata = true
+	exportConfig.SkipCredentials = true
+	exportConfig.SkipIPAddresses = true
+	exportConfig.SkipSSHHostKeys = true
+	exportConfig.SkipStatusHistory = true
+	exportConfig.SkipLinkLayerDevices = true
+
+	model, err := b.backend.ExportPartial(exportConfig)
+	if err != nil {
+		return params.BytesResult{}, common.ServerError(err)
+	}
+
+	err = model.Validate()
```

Comment:
Don't call Validate. It won't be valid. The ExportPartial is not defined to produce a valid model export.

Reply:
@howbazaar , agree that you have mentioned that in other PR that bundles are never accepted in any other format. the discussion started to have similar interface from server like its done in DumpModel command. Agree with your point and I have made it stringResult itself.

Reply:
All this versioning confusion started because i changed from StringResult to BytesResult. Its my fault i didn't take into consideration that gui makes use of this. Apologise.